nima/http/encoding/encoding/src/main/java/io/helidon/nima/http/encoding/ContentEncodingContext.java

```diff
@@ -88,4 +97,109 @@ static ContentEncodingContext create() {
      * @return content encoder to use
      */
     ContentEncoder encoder(Headers headers);
+
+    /**
+     * Builder to set up this encoding support content.
+     *
+     * @return a new builder
+     */
+    static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Fluent API builder for {@link ContentEncodingContext}.
+     */
+    class Builder implements io.helidon.common.Builder<Builder, ContentEncodingContext> {
+
+        private static final String IDENTITY_ENCODING = "identity";
+
+        private final HelidonServiceLoader.Builder<ContentEncodingProvider> encodingProviders
+                = HelidonServiceLoader.builder(ServiceLoader.load(ContentEncodingProvider.class));
+
+        // Force disable of encoders/decoders. Will build empty Maps when false.
+        private boolean enabled = true;
+
+        /**
+         * Update this builder from configuration.
+         * <p>
+         * Configuration:<ul>
+         *     <li><b>disable: true</b>  - to disable content encoding support</li>
+         * </ul>
+         *
+         * @param config configuration to use
+         * @return updated builder instance
+         */
+        public Builder config(Config config) {
+            config.get("disable").asBoolean().ifPresent(this::disableFromConfig);
```

Comment:
The config key should be `enabled`.
Also as we do not really want to handle this this way, the property should be `use-service-loader` or similar (this should already be used somewhere in Helidon).

Reply:
Changed to `discover-services` after some searching for existing implementations of this feature.